File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

void print_result(int i);
int calculate_index(int l, int w, int s);
int count_letters(string s);
int count_words(string s);
int count_sentences(string s);

int main(void)
{
    string input = get_string("Text: ");

    int letters = count_letters(input);
    int words = count_words(input);
    int sentences = count_sentences(input);

    int index = calculate_index(letters, words, sentences);

    print_result(index);
}

void print_result(int index)
{
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int calculate_index(int letters, int words, int sentences)
{
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    float index = (0.0588 * L) - (0.296 * S) - 15.8;

    int rounded = (int) round(index);

    return rounded;
}

int count_letters(string word)
{
    int count = 0;

    for (int i = 0, l = strlen(word); i < l; i++)
    {
        if (isalpha(word[i]))
        {
            count++;
        }
    }

    return count;
}

int count_words(string word)
{
    int count = 0;

    for (int i = 0, l = strlen(word); i < l; i++)
    {
        if (word[i] == ' ')
        {
            count++;
        }
    }

    return count + 1;
}

int count_sentences(string word)
{
    const int endings_length = 3;
    const char endings[endings_length] = {'.', '!', '?'};

    int count = 0;

    for (int i = 0, l = strlen(word); i < l; i++)
    {
        for (int j = 0; j < endings_length; j++)
        {
            if (word[i] == endings[j])
            {
                count++;
            }
        }
    }

    return count;
}

File: sentimental-readability/readability.py
def main():
    _input = input("Text: ")

    letter_count = count_letters(_input)
    word_count = count_words(_input)
    sentence_count = count_sentences(_input)

    index = calculate_index(letter_count, word_count, sentence_count)

    result = get_result(index)

    print(result)


def get_result(index):
    if (index >= 16):
        return "Grade 16+"

    if (index < 1):
        return "Before Grade 1"

    return f"Grade {index}"


def calculate_index(letters, words, sentences):
    l = letters / words * 100
    s = sentences / words * 100

    ret = (0.0588 * l) - (0.296 * s) - 15.8

    return round(ret)


def count_letters(text):
    return sum(c.isalpha() for c in text)


def count_words(text):
    return len(text.split(" "))


def count_sentences(text):
    return sum(text.count(ending) for ending in ['.', '!', '?'])


main()
